### PR TITLE
GH-38905: Spelling fixes

### LINKED ISSUE #38900: Spelling errors identified by check-spelling
### Describe the bug, including details regarding any error messages, version, and platform.

The [check-spelling action](https://github.com/marketplace/actions/check-spelling) enabled me to identify some misspellings...

The misspellings have been reported at https://github.com/jsoref/apache-arrow/actions/runs/7008141484/attempts/1#summary-19063787889

### Component(s)

C#, C++, C++ - Gandiva, FlightRPC, GLib, Go, Java, Parquet, Python, R, Ruby

### LINKED ISSUE #38905: Spelling errors identified by check-spelling
### Describe the bug, including details regarding any error messages, version, and platform.

The [check-spelling action](https://github.com/marketplace/actions/check-spelling) enabled me to identify some misspellings...

The misspellings have been reported at https://github.com/jsoref/apache-arrow/actions/runs/7008141484/attempts/1#summary-19063787889

### Component(s)

C#, C++, C++ - Gandiva, FlightRPC, GLib, Go, Java, Parquet, Python, R, Ruby

### PR DESCRIPTION
**Pull request by jsoref**
### Rationale for this change

If you make changes to any of the code generators (src/idl_gen*) be sure to
[build](https://google.github.io/flatbuffers/flatbuffers_guide_building.html) your project, as it will generate code based on the changes. If necessary
the code generation script can be directly run (scripts/generate_code.py),
requires Python3. This allows us to better see the effect of the PR.

I've made changes but haven't actually tried building it, sorry. I can do that later...

If your PR includes C++ code, please adhere to the
[Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html),
and don't forget we try to support older compilers (e.g. VS2010, GCC 4.6.3),
so only some C++11 support is available.

Technically yes, but I haven't changed anything meaningful.

Fixes misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/apache-arrow/actions/runs/7008141484/attempts/1#summary-19063787889

### Are these changes tested?

The action will report that the changes in this PR would make it happy: https://github.com/jsoref/apache-arrow/actions/runs/7008142287

### Are there any user-facing changes?

I hope not -- I'm happy to split any changes into a distinct PR if requested
* Closes: #38900
* Closes: #38905

### CODE REVIEW

File: js/src/visitor/builderctor.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export class GetBuilderCtor extends Visitor {
     public visitDurationSecond() { return DurationSecondBuilder; }
     public visitDurationMillisecond() { return DurationMillisecondBuilder; }
     public visitDurationMicrosecond() { return DurationMicrosecondBuilder; }
-    public visistDurationNanosecond() { return DurationNanosecondBuilder; }
```
Comment (domoritz) [inline js/src/visitor/builderctor.ts:99]: It's probably not minor anymore.

**Review by domoritz — APPROVED** [review]
I checked the Js changes. Look good.

**Review by assignUser — CHANGES_REQUESTED** [review]
This is not a minor PR (max 2 files touched && no functional changes) please create an issue

File: ruby/red-arrow/lib/arrow/sort-key.rb [hunk 2]
```diff
@@ -79,9 +79,9 @@ def try_convert(value)
     #     target and corresponding order is used. `"+"` uses ascending
     #     order and `"-"` uses ascending order.
     #
-    #     If `target` is not a String nor `target` doesn't start with the
-    #     leading order mark, sort column target is `target` as-is and
-    #     ascending order is used.
+    #     If `target` is either not a String or `target` doesn't start
+    #     with the leading order mark, sort column is `target` as-is
+    #     and ascending order is used.
```
Comment (jsoref) [inline ruby/red-arrow/lib/arrow/sort-key.rb:84]: notable

**Comment by kou** [comment]
This is too large to review.
Could you split this to small PRs?
For example, you can open one PR per subdirectory such as `.github/`, `c_glib/` and `ci/`.
But one PR per subdirectory may be still large. For example, one PR for `cpp/` will be large. You can open one PR per subsub*directory such as `cpp/src/arrow/acero/` and `cpp/src/arrow/array/`.

**Comment by AlenkaF** [comment]
If I am not mistaken the changes got differed to smaller PRs and we can close this one and the associated issue left open, right @jsoref?

**Comment by jsoref** [comment]
Ok

**Pull request closed without merging**